### PR TITLE
Merge remote-tracking branch 'origin/upstream_sql' into 'sql'

### PR DESCRIPTION
**Pull request by buddhiayesha2015**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the commit message is formatted like:
   `[GEARPUMP-<Jira issue #>] Meaningful description of pull request` 
 - [ ] Make sure tests pass via `sbt clean test`.
 - [ ] Make sure old documentation affected by the pull request has been updated and new documentation added for new functionality.

### CODE REVIEW

File: examples/streaming/wordcount-java/src/test/java/org/apache/gearpump/streaming/examples/wordcountjava/WordCountSpecJava.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.apache.gearpump.streaming.examples.wordcountjava.dsl.WordCount;
+import org.apache.log4j.Logger;
+import org.junit.Test;
+
+/**
+ * Created by Buddhi on 7/14/2017.
+ */
+public class WordCountSpecJava {
+
+    private final static Logger logger = Logger.getLogger(WordCountSpecJava.class);
```
Comment (manuzhang) [inline examples/streaming/wordcount-java/src/test/java/org/apache/gearpump/streaming/examples/wordcountjava/WordCountSpecJava.java]: We indent with 2 spaces.

File: examples/streaming/wordcount-java/src/test/java/org/apache/gearpump/streaming/examples/wordcountjava/WordCountSpecJava.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import org.apache.gearpump.cluster.ClusterConfig;
+import org.apache.gearpump.streaming.examples.wordcountjava.dsl.WordCount;
+import org.apache.log4j.Logger;
+import org.junit.Test;
+
+/**
+ * Created by Buddhi on 7/14/2017.
+ */
+public class WordCountSpecJava {
```
Comment (manuzhang) [inline examples/streaming/wordcount-java/src/test/java/org/apache/gearpump/streaming/examples/wordcountjava/WordCountSpecJava.java]: Is this used anywhere ? It's a simple wrapper around `org.apache.gearpump.streaming.examples.wordcountjava.dsl.WordCount`

Comment (buddhiayesha2015) [inline examples/streaming/wordcount-java/src/test/java/org/apache/gearpump/streaming/examples/wordcountjava/WordCountSpecJava.java]: I will remove that. That is a testing class and it is not essential

File: experiments/sql/src/main/java/org/apache/gearpump/sql/rel/GearAggregationRel.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import org.apache.calcite.util.Util;
+import org.apache.gearpump.sql.table.SampleString;
+import org.apache.gearpump.streaming.dsl.api.functions.MapFunction;
+import org.apache.gearpump.streaming.dsl.api.functions.ReduceFunction;
+import org.apache.gearpump.streaming.dsl.javaapi.JavaStream;
+import org.apache.gearpump.streaming.dsl.javaapi.JavaStreamApp;
+import org.apache.gearpump.streaming.dsl.javaapi.functions.GroupByFunction;
+import org.apache.gearpump.streaming.dsl.window.api.Trigger;
+import org.apache.gearpump.streaming.dsl.window.api.WindowFunction;
+import org.apache.log4j.Logger;
```
Comment (manuzhang) [inline experiments/sql/src/main/java/org/apache/gearpump/sql/rel/GearAggregationRel.java]: We use `org.slf4j.Logger` and `Logger  LOG = LogUtil.getLogger(className)` for logging

Comment (buddhiayesha2015) [inline experiments/sql/src/main/java/org/apache/gearpump/sql/rel/GearAggregationRel.java]: I will fix it.

**Comment by manuzhang** [comment]
LGTM

**Comment by manuzhang** [comment]
@buddhiayesha2015 The PR has been merged. You may close it now (it cannot be automatically closed for merging into feature branch)

**Pull request closed without merging**